packages/core/src/App/Components/Routes/binary-routes.jsx

```diff
@@ -4,23 +4,34 @@ import { Loading } from '@deriv/components';
 import getRoutesConfig from 'App/Constants/routes-config';
 import RouteWithSubRoutes from './route-with-sub-routes.jsx';
 import { observer, useStore } from '@deriv/stores';
-import { getPositionsV2TabIndexFromURL, isDTraderV2, routes } from '@deriv/shared';
+import { getPositionsV2TabIndexFromURL, isDTraderV2, isDTraderV2Width, routes } from '@deriv/shared';
+import { useGrowthbookGetFeatureValue } from '@deriv/hooks';
 
 const BinaryRoutes = observer(props => {
     const { ui, gtm } = useStore();
     const { promptFn, prompt_when } = ui;
     const { pushDataLayer } = gtm;
     const location = useLocation();
-    const is_dtrader_v2 =
-        isDTraderV2() && (location.pathname.startsWith(routes.trade) || location.pathname.startsWith('/contract/'));
+    const [dtrader_v2_enabled_gb] = useGrowthbookGetFeatureValue({
+        featureFlag: 'dtrader_v2_enabled',
+        defaultValue: false,
+    });
+    const [dtrader_v2_enabled, setDTraderV2Enabled] = React.useState(false);
+    React.useEffect(() => {
+        setDTraderV2Enabled(
+            (isDTraderV2() || (Boolean(dtrader_v2_enabled_gb) && isDTraderV2Width())) &&
+                (location.pathname.startsWith(routes.trade) || location.pathname.startsWith('/contract/'))
+        );
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [dtrader_v2_enabled_gb]);
```

Comment:
I see almost the same logic in multiple places. 
Would it make sense to contain that logic into one function or one hook to be used everywhere?
especially given that the logic is non-trivial and quite complex,

Reply:
I have addressed this and put it inside a hook